oops/backplane/unittester_support.py

```diff
@@ -0,0 +1,389 @@
+
+    This function must be called by main() in order for these arguments to
+    to be parsed.  If not, the above default values will apply."""
+    import argparse
+    import sys
+
+    ## Define arguments ##
+    parser = argparse.ArgumentParser(description='Backplane unit tester.')
+
+
+    # Generic arguments
+    parser.add_argument('--args', nargs='*', metavar='arg', default=None,
+                        help='Generic arguments to pass to the test modules. '
+                             'Must occur last in the argument list.')
+
+
+    # Basic controls
+    parser.add_argument('--verbose', action='store_true', default=None,
+                        help='Print output to the terminal.')
+
+#    parser.add_argument('--diff', nargs=2, metavar=('old', 'new'), default=None,
+#                        help='Compare new and old backplane logs.')
+#
+    parser.add_argument('--exercises-only', action='store_true', default=None,
+                        help='Execute only the backplane exercises.')
+
+    parser.add_argument('--no-exercises', action='store_true', default=None,
+                        help='Execute all tests except the backplane exercises.')
+
+    parser.add_argument('--no-compare', action='store_true', default=None,
+                        help='Do not compare backplanes with references.')
+
+    parser.add_argument('--output', nargs=1, metavar='dir', default=None,
+                        help='Directory in which to save backplane PNG images. '
+                             'Default is TESTDATA_PARENT_DIRECTORY/[data dir]. '
+                             'If the directory does not exist, it is created.')
+
+    parser.add_argument('--no-output', action='store_true', default=None,
+                        help='Disable saving of backplane PNG files.')
+
+    parser.add_argument('--log', action='store_true', default=None,
+                        help='Enable the internal oops logging.')
+
+    parser.add_argument('--undersample', nargs=1, type=int, metavar='N', default=None,
+                        help='Amount by which to undersample backplanes.  Default is 16.')
+
+    parser.add_argument('--reference', action='store_true', default=None,
+                        help='Generate reference backplanes and exit.')
+
+### TODO: currently only works for 80-char width
+    parser.add_argument('--test-level', nargs=1, type=int, metavar='N', default=None,
+                        help='Selects among pre-set parameter combinations:  '
+                              '-test_level 1: no printing, no saving, undersample 32. '
+                              '-test_level 2: printing, no saving, undersample 16. '
+                              '-test_level 3: printing, saving, no undersampling. '
+                              'These behaviors are overridden by other arguments.')
+
+
+    ## Parse arguments, leaving unknown args for some other parser ##
+    argv, left = parser.parse_known_args()
```

Comment:
It's confusing that this variable is called `argv` when it isn't actually the `argv` or any part of it, it's the result of the argument parsing. Perhaps rename to something like `args`.

Reply:
Changed ARGS to GENERIC_ARGS and ARGV to ARGS.